app_dart/lib/src/datastore/cocoon_config.dart

```diff
@@ -71,8 +71,18 @@ class Config {
     return String.fromCharCodes(cacheValue).split(',');
   }
 
-  Future<List<LuciBuilder>> getRepoLuciBuilders(String bucket, String repo) async {
```

Comment:
Cocoon config is designed to wrap operations to Cocoon datastore. These functions have nothing to do with the datastore, therefore it's confusing to see them.

Reply:
Cocoon config has now been extended to support functions not limited to datastore, including github service, gold status refresh, etc. Nevertheless, all these are configs related to cocoon.
It might be better to move `cocoon_config.dart` out of existing `datastore` folder to avoid confusion, or rename `datastore`. 
Created issue https://github.com/flutter/flutter/issues/64666 to track.

Reply:
`cocoon_config.dart` is now the central place to manage all configs related to `cocoon` backend server, not limited to the `Cocoon` repo. 
[`githubClient`](https://github.com/flutter/cocoon/blob/master/app_dart/lib/src/datastore/cocoon_config.dart#L263), [`cirrusGraphQLClient`](https://github.com/flutter/cocoon/blob/master/app_dart/lib/src/datastore/cocoon_config.dart#L290), etc. are similarly servicing other repos.

When you talk about complexity, I guess you are referring to the length or function numbers. If that is the case, we may need to consider to split the whole config file to sub ones.

Reply:
Let's move the discussion to the bug. Ideally we still want to have a single object that we can pass around to get the configurations. However we can do better on splitting the functionality as luci_configs, github_configs, etc.